src/analysis/processing/qgsmetadataalgorithms.cpp

```diff
@@ -0,0 +1,252 @@
+/***************************************************************************
+                         qgsmetadataalgorithms.cpp
+                         ---------------------
+    begin                : November 2024
+    copyright            : (C) 2024 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmetadataalgorithms.h"
+
+///@cond PRIVATE
+
+QString QgsCopyLayerMetadataAlgorithm::name() const
+{
+  return QStringLiteral( "copylayermetadata" );
+}
+
+QString QgsCopyLayerMetadataAlgorithm::displayName() const
+{
+  return QObject::tr( "Copy layer metadata" );
+}
+
+QStringList QgsCopyLayerMetadataAlgorithm::tags() const
+{
+  return QObject::tr( "change,layer,metadata,qmd" ).split( ',' );
+}
+
+QString QgsCopyLayerMetadataAlgorithm::group() const
+{
+  return QObject::tr( "Layer tools" );
```

Comment:
What do you think about making a new dedicated group for metadata?

Reply:
I was not sure about adding dedicated group as initially it should be just one tool :-). But with 3 tools and more tools planned it definitely makes sense.